### DISCUSSION TITLE
[Azure DevOps] on-premise support

### DISCUSSION THREAD

**Discussion by simonrrr** [apache/devlake#7259] [General]
Hi! Is there any reason that Azure DevOps is hard-coded to use the cloud URL (dev.azure.com)? In the UI there’s a greyed out ‘not supported’ for on-premise / non-cloud Azure DevOps - but that should be incredibly easy to support by just changing the API URL to the on-premise hosted version. We found exactly the same with Backstage and had to (easily) hack the code to support it.

**Comment by klesh** [comment]
Yes, you may be correct, but we don't have an on-premise Azure DevOps to test against to 😂.
It would be nice if you have one and are willing to work on it. 😄
